src/utils/openwhisk.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2018 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-disable camelcase,no-underscore-dangle */
+const querystring = require('querystring');
+const owwrapper = require('@adobe/openwhisk-loggly-wrapper');
+
+/**
+ * A function that takes OpenWhisk-style req parameter and turns
+ * it into the original Express-style request object which is returned.
+ * @param {Object} action The pipeline action context
+ * @return {Object} an express style request object.
+ */
+function extractClientRequest(action) {
+  const { request } = action;
+  try { // the edge encodes the client request parameters into the `params` param ;-)
+    if (!request || !request.params) {
+      return {};
+    }
+    if (request.params.params) {
+      return {
+        params: querystring.parse(request.params.params),
+        headers: request.headers,
```

Comment:
Wouldn't this be in `__ow_headers`?

Reply:
I see that `extractActionContext ` is run beforehand now and changes `__ow_headers` to just `headers`. You might want to clarify that in the JSDoc, as the `action` object is not entirely-OW-style.

Reply:
it is already:

```
 * @param {Object} action The pipeline action context
```